### ISSUE TITLE
Check if an image is a gif

### ISSUE THREAD

**Issue by touzainanboku** [ccrama/Slide#2232]
Slide version: 5.5-alpha2-ditty
Android version: 6.0.1

Some links like [this one](http://i.imgur.com/DFdwqHP.jpg) have the incorrect extension in the URL, where it says .jpg when it's actually a gif. It'd be great if Slide could check the file type regardless of what the URL says and then act accordingly.

**Comment by Nxt3** [comment]
I don't think this is possible--but I'm not 100% sure.

**Comment by touzainanboku** [comment]
I've seen other apps do it, so it's not entirely impossible. Although whether it's possible with Slide's implementation is a different matter, so I understand if it's not feasible.

**Comment by Alexendoo** [comment]
It will still display correctly when opened for most cases of mismatches content type and extensions, however the tag will be wrong before opening the content. To accurately detect for the tag would require making an external HTTP request for every link which isn't really feasible

**Comment by touzainanboku** [comment]
I don't really mind the tag. I'm only talking about after it's opened.

**Comment by Alexendoo** [comment]
For opening it should work fine, jpg → gif at least, if there's a post where it isn't could you please grab a link to it

**Comment by touzainanboku** [comment]
The image I linked in the op doesn't work. (Here)[https://reddit.com/r/funny/comments/50s9c0/cat_reunited_after_2_months_of_being_lost/?ref=search_posts]'s a submission with the same link.

**Comment by Alexendoo** [comment]
Strange that one doesn't work, some do

**Comment by fbis251** [comment]
The Imgur API gives you 12,500 API requests per day, how close are you getting with just albums? If you still have some calls to spare you should use their image endpoint to get the actual file type and other info for images/mp4s

**Comment by Alexendoo** [comment]
We have the open source exception, I believe. However the mime type is included in the response, I think it's just not checked first thing

**Comment by ccrama** [comment]
The issue is with data saving, with data saving turned off it checks, but with data saving turned on, it loads the Reddit static image

**Comment by touzainanboku** [comment]
Makes sense. I've seen that in most cases where a static image is displayed when it's actually a gif, tapping HQ loads the gif.

In the link I posted however, I still got a static image even with data saving off.
